### PR TITLE
Vector client release 0.4.0

### PR DESCRIPTION
**Pull request by DomPeliniAerospike**
VEC-94
VEC-95
VEC-96
VEC-98
VEC-103
VEC-107
VEC-108
VEC-109
VEC-123

### CODE REVIEW

**Review by justinlee-aerospike — CHANGES_REQUESTED** [review]
Generally looks okay, but a few comments:

- proto files: These are shared between clients, how are you going to keep them in sync?  Should this be submoduled out?  
- Python 3.8 goes end of life this year - perhaps you should move the min Python to 3.9
- main README file needs to be changed - PyPi upload instructions should be removed if this is to be customer facing

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+ignore = E501
+max-line-length = 88 # black compatability
+ignore = E501, E203  # black compatability
```
Comment (jdogmcsteezy) [inline .flake8:4]:
```suggestion
max-line-length = 88 # black compatability
ignore = E501, E203  # black compatability
```

**Review by jdogmcsteezy — APPROVED** [review]
I mostly looked at the recent commits because I already reviewed this PR. A couple nit picks but otherwise LGTM

**Comment by jdogmcsteezy** [comment]
@justinlee-aerospike, submoduling is a good idea. I am drawing a blank on what other options we have other than simply copying them over

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+import os
+import sys
+sys.path.insert(0, os.path.abspath('../src/aerospike_vector'))
+import sphinx.ext.autodoc
+
+project = 'D'
```
[the remark this answers is not in the document]
Comment (DomPeliniAerospike) [reply docs/conf.py:13]: Yeah this was just a place holder while I was getting the hang of the python API docs.
Will fix this with the (next) docs PR.

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.. D documentation master file, created by
+   sphinx-quickstart on Sun Apr  7 22:14:55 2024.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to D's documentation! SCOBEDY BABALOUIE
```
[the remark this answers is not in the document]
Comment (DomPeliniAerospike) [reply docs/index.rst:6]: I did not need to write this, will fix this soon lol.

**Comment by DomPeliniAerospike** [comment]
> @justinlee-aerospike, submoduling is a good idea. I am drawing a blank on what other options we have other than simply copying them over

Submoduling needs to happen for the future, probably not a priority for this release.

**Pull request merged**